src/main/query_profiler.cpp

```diff
@@ -130,22 +128,20 @@ bool QueryProfiler::OperatorRequiresProfiling(PhysicalOperatorType op_type) {
 }
 
 void QueryProfiler::Finalize(ProfilingNode &node) {
-	auto &op_node = node.Cast<OperatorProfilingNode>();
-
-	for (idx_t i = 0; i < op_node.GetChildCount(); i++) {
-		auto child = op_node.GetChild(i);
+	for (idx_t i = 0; i < node.GetChildCount(); i++) {
+		auto child = node.GetChild(i);
 		Finalize(*child);
-		if (op_node.type == PhysicalOperatorType::UNION &&
-		    op_node.GetProfilingInfo().Enabled(MetricsType::OPERATOR_CARDINALITY)) {
-			op_node.GetProfilingInfo().AddToMetric(
+		auto type = PhysicalOperatorType(node.GetProfilingInfo().GetMetricValue<uint8_t>(MetricsType::OPERATOR_TYPE));
```

Comment:
Nit for readability: should we add a line with `auto &info = node.GetProfilingInfo()`?